src/main/java/com/example/GymProject/dao/TraineeDao.java

```diff
@@ -3,85 +3,69 @@
     }
+    private final ResourceLoader resourceLoader;
 
-    public void create(Trainee trainee){
-        try {
-            if (trainee.getUserId() == null) {
-                logger.warning("Trainee userId is null");
-                throw new IllegalArgumentException("UserId cannot be null");
-            }
-            traineeMap.put(trainee.getUserId(), trainee);
-            logger.info("Inserted New Trainee");
-        } catch (Exception e){
-          logger.severe("Error While Inserting New Value");
-            throw new IllegalArgumentException(e);
-        }
+    public TraineeDao(ResourceLoader resourceLoader) {
+        this.resourceLoader = resourceLoader;
     }
 
-    public Trainee select(String key){
+    public void create(Trainee trainee) {
+
+        Assert.notNull(trainee.getUserId(), "UserId cannot be null");
+        traineeMap.put(trainee.getUserId(), trainee);
+        logger.info("Inserted New Trainee");
+
+    }
+
+    public Trainee select(String key) {
+
         Trainee trainee;
-        try {
-            if(traineeMap.containsKey(key)){
-                logger.info("Trainee found");
-                trainee = traineeMap.get(key);
-            }else{
-               logger.warning("Wrong Key, Trainee not Found");
-                throw new IllegalArgumentException("Trainee Not Found");
-            }
-            return trainee;
-        } catch (Exception e){
-            throw new IllegalArgumentException(e);
-        }
+        Assert.isTrue(traineeMap.containsKey(key), "Trainee Not Found");
+        logger.info("Trainee found");
+        trainee = traineeMap.get(key);
+
+        return trainee;
+
     }
 
-    public void update(String key, Trainee trainee){
-        try {
-            if (key == null) {
-               logger.warning("Trainee userId is null");
-                throw new IllegalArgumentException("UserId cannot be null");
-            }
-            traineeMap.put(key, trainee);
-            logger.info("Trainee Updated");
-        } catch (Exception e){
-          logger.warning("Trainee has not been Updated");
-            throw new IllegalArgumentException(e);
-        }
+    public void update(String key, Trainee trainee) {
+        Assert.notNull(key,"UserId cannot be null");
+        Assert.notNull(trainee,"The trainee parameter cannot be null");
+        traineeMap.put(key, trainee);
+        logger.info("Trainee Updated");
+
     }
-    public void delete(String key){
-        try {
-            if (traineeMap.containsKey(key)){
-                traineeMap.remove(key);
-                logger.info("Trainee Removed Successfully!");
-            }else {
-               logger.warning("Trainee has not been Removed");
-                throw new IllegalArgumentException("Wrong Key");
-            }
-        } catch (Exception e){
-            throw new IllegalArgumentException(e);
-        }
+
+    public void delete(String key) {
+        Assert.notNull(key,"UserId cannot be null");
```

Comment:
"The key parameter not to be null". What if this key does not represent userId any more?

Reply:
Or change your signature in a way to rename key -> userId

Reply:
`public void delete(String userId)`